### ISSUE TITLE
route53 mx records are always reported as changed

### ISSUE THREAD

**Issue by pedrofaustino** [ansible/ansible-modules-core#992]
**Issue Type**: “Bug Report”

**Ansible Version**: 

$ ansible --version
ansible 1.8.3
  configured module search path = /usr/local/etc/ansible

**Environment**: OSX 10.10.2

**Summary**:

When running the following route53 task, the result is always "changed", even if the record is already set.

```
    - name: Google MX
      route53:
        command: create
        zone: example.com
        record: example.com
        type: MX
        value: '10 aspmx.l.google.com, 20 alt1.aspmx.l.google.com, 20 alt2.aspmx.l.google.com, 30 aspmx2.googlemail.com, 30 aspmx3.googlemail.com, 30 aspmx4.googlemail.com, 30 aspmx5.googlemail.com'
        overwrite: true
```

**Steps To Reproduce**:

Create a playbook with that task, and call it with 
`ansible-playbook -i 'localhost,' playbook.yml --connection=local`

**Expected Results**:

I was expecting to have a "ok" result, green.

**Actual Results**:

I'm getting "changed", yellow.

**Comment by moorereason** [comment]
I suspect that your issue is the space after the commas in your `value` field.  The module should probably be modified [from this](https://github.com/ansible/ansible-modules-core/blob/d960dc1584fb17ebdb720c0c65f7ae155ab612fa/cloud/amazon/route53.py#L227):

``` python
value_list = sorted(value_in.split(','))
```

to this:

``` python
value_list = sorted([s.strip() for s in value_in.split(',')])
```

I prefer your spacing since it's easier to read, but a simple workaround is to use a list instead of a string in your task:

``` yaml
value: ['10 aspmx.l.google.com', '20 alt1.aspmx.l.google.com', '20 alt2.aspmx.l.google.com', '30 aspmx2.googlemail.com', '30 aspmx3.googlemail.com', '30 aspmx4.googlemail.com', '30 aspmx5.googlemail.com']
```

or even more readable:

``` yaml
value: [
  '10 aspmx.l.google.com',
  '20 alt1.aspmx.l.google.com',
  '20 alt2.aspmx.l.google.com',
  '30 aspmx2.googlemail.com',
  '30 aspmx3.googlemail.com',
  '30 aspmx4.googlemail.com',
  '30 aspmx5.googlemail.com'
]
```

**Comment by jonhy81** [comment]
Hi, @moorereason. Thanks for your help. worked like a charm.

**Comment by bcoca** [comment]
closed via merged #1576
